docs/DeveloperGuide.md

```diff
@@ -389,144 +363,284 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
     
 <br>
 
-**Use case 05: Delete a task**
+
+
+**Use case: UC03 - List all tasks**
 
 **MSS**
 
 1. User requests to list tasks.
 
 2. PlaNus shows a list of tasks.
 
-3. User requests to delete a specific task from the list.
-
-4. PlaNus deletes the task.
-
   Use case ends.
 
 **Extensions**
 
-- 2a. The list is empty.
+  - 2a. The list is empty.
+
+    Use case ends.
+
+<br>
+
+**Use case: UC04 - Mark deadlines as done**
+
+**MSS**
+
+1. User requests to list tasks.
+
+2. PlaNus shows a list of tasks.
+
+3. User requests to mark certain deadlines in the list as done.
+
+4. PlaNus marks the deadline as done.
 
   Use case ends.
 
+**Extensions**
+
 - 3a. The given index is invalid.
 
   - 3a1. PlaNus shows an error message.
 
     Use case resumes at step 3.
 
+- 3b. Some indexes given are not of a deadline.
+
+  - 3b1. PlaNus shows an error message.
+
+    Use case resumes at step 3.
+
+- 3c. Deadlines represented by some indexes are already marked as completed.
+
+  - 3c.1 PlaNus shows an error message.
+
+    Use case resumes at step 3.
+
 <br>
 
-**Use case 06: List all tasks**
+**Use case: UC05 - Find tasks**
 
 **MSS**
 
-1. User requests to list tasks.
+1. User <ins>lists a set of tasks (UC03 or UC05)</ins>
 
-2. PlaNus shows a list of tasks.
+2. User requests to find tasks with specified search phrase(s) in specified attribute(s).
+
+3. PlaNus shows a list of tasks that match the specified search phrase(s) in the specified attribute(s).
 
   Use case ends.
 
 **Extensions**
 
-  - 2a. The list is empty.
+- 2a. The search phrase is empty or consists of only white spaces.
+
+  - 2a1. PlaNus shows an error message.
+
+    Use case ends.
+
+- 2b. The search phrase is in invalid format or includes invalid characters.
+
+    - 2b1. PlaNus shows an error message.
 
     Use case ends.
 
+- 3a. The list is empty.
+
+  Use case ends.
+
 <br>
 
-**Use case 07: Request help**
+**Use case: UC06 - Edit a task**
+
+**Guarantees**
+
+* A task will be edited only if the task is still valid after being edited.
```

Comment:
what does still valid mean? Maybe can specify that the task must not overlap with existing tasks or lessons after being edited